api/spec/src/billing/invoice/bill.tsp

```diff
@@ -0,0 +1,407 @@
+ * ordering: Ordering details including document references and buyer or seller parties.
+ *    -> We are not supporting this yet (e.g. order handled by seperate entity, physical goods delivery details, etc.)
+ * delivery: Specific details on delivery of the goods referenced in the invoice.
+ *    -> We are not supporting physical goods delivery yet.
+ * tax: Special tax configuration for billing.
+ *    -> The prices include is the meaningful part, but we should make that an array, let's see if we need this as we implement more features.
+ */
+@summary("Invoice represents a document that is sent to a customer to request payment for goods or services.")
+@friendlyName("BillingInvoice")
+model Invoice {
+  ...InvoiceSummary;
+
+  @summary("The taxable entity supplying the goods or services.")
+  @visibility("read", "query") // Seperate API endpoint is available for updating
+  supplier: Org.Party;
+
+  @summary("Legal entity receiving the goods or services.")
+  @visibility("read", "query") // Seperate API endpoint is available for updating
+  customer: Org.Party;
+
+  @summary("List of invoice lines representing each of the items sold to the customer.")
+  @visibility("read", "query") // Seperate API endpoint is available for updating
+  lines?: Line[];
+
+  @summary("Discounts or allowances applied to the complete invoice.")
+  @visibility("read", "query") // Seperate API endpoint is available for updating
+  discounts?: Discount[];
+
+  @summary("Information on when, how, and to whom the invoice should be paid.")
+  @visibility("read", "query") // Seperate API endpoint is available for updating
+  payment?: Payment;
+}
+
+@friendlyName("BillingInvoceType")
+enum InvoiceType {
+  @summary("A regular commercial invoice document between a supplier and customer.")
+  standard: "standard",
+
+  @summary("For a clients validation before sending a final invoice.")
+  proforma: "proforma",
+
+  @summary("Corrected invoice that completely *replaces* the preceding document.")
+  corrected: "corrected",
+
+  @summary("Reflects a refund either partial or complete of the preceding document. A credit note effectively *extends* the previous document.")
+  credit_note: "credit-note",
+
+  @summary("An additional set of charges to be added to the preceding document.")
+  debit_note: "debit-note",
+
+  @summary("Any other type of invoice that does not fit into the standard categories and implies\nthat any scenarios defined in tax regimes or addons will not be applied.\n\nThis is useful for being able to create invoices with custom types in extensions, but is not recommend for general use.")
+  other: "other",
+}
+
+@friendlyName("BillingInvoiceDiscount")
+@summary("Discount represents an allowance applied to the complete document independent from the individual lines.")
+model Discount {
+  @summary("Universally Unique Identifier.")
+  @format("uuid")
+  uuid?: string;
+
+  @summary("Line number inside the list of discounts (calculated)")
+  i?: int32;
```

Comment:
how about `lineNumber`?

Reply:
I have remove this altogether, as I hope we wouldn't need this due to having ULIDs or some stable sorting. If we end up needing this I will re-add.